smithy-model/src/test/java/software/amazon/smithy/model/loader/ModelAssemblerTest.java

```diff
@@ -391,6 +391,26 @@ public void detectsMetadataConflicts() {
                    containsString("Metadata conflict for key `foo`"));
     }
 
+    @Test
+    public void metadataIsNotAffectedByTheSourceName() {
+        Model model1 = new ModelAssembler()
+                .addUnparsedModel("a1.smithy", "metadata items = [1]")
+                .addUnparsedModel("a2.smithy", "metadata items = [2]")
+                .addUnparsedModel("a3.smithy", "metadata items = [3]")
+                .assemble()
+                .unwrap();
+        Model model2 = new ModelAssembler()
+                .addUnparsedModel("b1.smithy", "metadata items = [1]")
+                .addUnparsedModel("b2.smithy", "metadata items = [2]")
+                .addUnparsedModel("b3.smithy", "metadata items = [3]")
+                .assemble()
+                .unwrap();
+        List<Number> metadata1 = model1.getMetadata().get("items").expectArrayNode().getElements().stream().map(s -> s.expectNumberNode().getValue()).toList();
```

Comment:
The test is failing because we currently support Java 8 :(

This will need to use `.collect(Collectors.toList())`